object_manager.go

```diff
@@ -441,6 +448,127 @@ func (objMgr *ObjectManager) CreateEADefinition(eadef EADefinition) (*EADefiniti
 	return newEadef, err
 }
 
+func (objMgr *ObjectManager) CreateIPv6Network(netview string, cidr string, name string) (*IPv6Network, error) {
+
+	ea := objMgr.getBasicEA(true)
+	ipv6network := NewIPv6Network(netview, cidr, ea)
+
+	if name != "" {
+		ipv6network.Ea["Network Name"] = name
+	}
+	ref, err := objMgr.connector.CreateObject(ipv6network)
+	if err != nil {
+		return nil, err
+	}
+	ipv6network.Ref = ref
+
+	return ipv6network, nil
+}
+
+func (objMgr *ObjectManager) GetIPv6Network(netview string, cidr string, ea EA) (*IPv6Network, error) {
+	if netview != "" && cidr != "" {
+		var res []IPv6Network
+
+		ipv6Network := NewIPv6Network(netview, cidr, ea)
+
+		if cidr != "" {
+			ipv6Network.Cidr = cidr
+		}
+
+		if ea != nil && len(ea) > 0 {
+			ipv6Network.eaSearch = EASearch(ea)
+		}
+
+		err := objMgr.connector.GetObject(ipv6Network, "", &res)
+
+		if err != nil || res == nil || len(res) == 0 {
```

Comment:
If err == nil but res == nil or len(res) == 0 as well, then the error returned will be nil also, and this is a problem. You should distinguish these cases, and in case when err == nil but we have an error by some other reason, you have to generate your own error message and return it.